frontend/src/upload.js

```diff
@@ -1,74 +1,142 @@
+} else {
+  window.referrer = 'external';
+}
+
 $(document).ready(function() {
-  gcmCompliant().catch(err => {
-    $('#page-one').attr('hidden', true);
-    $('#unsupported-browser').removeAttr('hidden');
-  });
-
-  $('#file-upload').change(onUpload);
-  $('body').on('dragover', allowDrop).on('drop', onUpload);
-  // reset copy button
-  const $copyBtn = $('#copy-btn');
-  $copyBtn.attr('disabled', false);
-  $('#link').attr('disabled', false);
-  $copyBtn.attr('data-l10n-id', 'copyUrlFormButton');
-
-  if (localStorage.length === 0) {
-    toggleHeader();
-  } else {
-    for (let i = 0; i < localStorage.length; i++) {
-      const id = localStorage.key(i);
-      //check if file exists before adding to list
-      checkExistence(id, true);
+  if (!location.pathname.toString().includes('download')) {
+    gcmCompliant()
+      .catch(err => {
+        $('#page-one').attr('hidden', true);
+        $('#unsupported-browser').removeAttr('hidden');
+        // record unsupported event
+        window.analytics
+              .sendEvent('sender', 'unsupported', {
+                cd6: err
+              });
+    });
+
+    $('#file-upload').change(onUpload);
+
+    $('.legal-links a, .social-links a, #dl-firefox').click(function(target) {
+      target.preventDefault();
+      const metric = findMetric(target.currentTarget.href);
+      // record exited event by recipient
+      window.analytics
+            .sendEvent('sender', 'exited', {
+              cd3: metric
+            })
+            .then(() => {
+              location.href = target.currentTarget.href;
+            });
+    })
+
+    $('#send-new-completed').click(function(target) {
+      target.preventDefault();
+      // record restarted event 
+      window.analytics
+            .sendEvent('sender', 'restarted', {
+              cd2: 'completed'
+            })
+            .then(() => {
+              localStorage.setItem('referrer', 'completed-upload');
+              location.href = target.currentTarget.href;
+            });
+    })
+
+    $('#send-new-error').click(function(target) {
+      target.preventDefault();
+      // record restarted event
+      window.analytics
+            .sendEvent('sender', 'restarted', {
+              cd2: 'errored'
+            })
+            .then(() => {
+              localStorage.setItem('referrer', 'errored-upload');
+              location.href = target.currentTarget.href;
+            });
+    })
+
+    $('body').on('dragover', allowDrop).on('drop', onUpload);
+    // reset copy button
+    const $copyBtn = $('#copy-btn');
+    $copyBtn.attr('disabled', false);
+    $('#link').attr('disabled', false);
+    $copyBtn.attr('data-l10n-id', 'copyUrlFormButton');
+
+    if (localStorage.length === 0) {
+      toggleHeader();
+    } else {
+      for (let i = 0; i < localStorage.length; i++) {
+        const id = localStorage.key(i);
+        //check if file exists before adding to list
+        checkExistence(id, true);
```

Comment:
... Because I notice we have code like this in ~4 places currently:

```diff
+          if (id !== 'totalUploads' && 
+              id !== 'totalDownloads' &&
+              id !== 'referrer') {
+            // ...
+          }
```

Maybe we need some utility `function isFile(id) {}` helper.

```js
function isFile(id) {
  return !['referrer', 'totalDownloads', 'totalUploads'].contains(id); // optionally validate `id` against a RegExp.
}